src/app/util/layer_utils.h

```diff
@@ -14,13 +14,19 @@ namespace doc {
 
 namespace app {
 
+  class Editor;
+
   // Calculates a possible candidate to be selected in case that we
   // have a specific "selectedLayer" and are going to delete the given
   // "layerToDelete".
   doc::Layer* candidate_if_layer_is_deleted(
     const doc::Layer* selectedLayer,
     const doc::Layer* layerToDelete);
 
+  // True if the active layer is locked (itself or its hierarchy),
+  // also, it sends a tip to the user 'Layer ... is locked'
+  bool layer_is_locked(Editor* editor);
```

Comment:
I would have used a `Layer* layer` as parameter, but it's fine, we can use the `Editor` (and change it later if it's needed, e.g. if we need to use this function from a command where there is no `Editor`)